### PR TITLE
Update test_pass_failures notebook to match metric template + include automation bits

### PR DESCRIPTION
**Pull request by hemajv**
Co-authored-by: Hema Veeradhi <hveeradh@redhat.com>
Co-authored-by:  Oindrilla Chatterjee <ochatter@redhat.com>

## Related Issues and Dependencies
#185 #161 

## This introduces a breaking change

- [ ] Yes
- [X] No

## Description

Converting the `test_pass_failures` notebook to follow the metric template

### CODE REVIEW

**Comment by oindrillac** [comment]
@harshad16 do you know why the Build Check is failing here?

**Comment by harshad16** [comment]
The build failing due to this:
```
ERROR: Could not find a version that satisfies the requirement ipykernel==5.5.2
ERROR: No matching distribution found for ipykernel==5.5.2
Failed to install requirements, dependency '\''ipykernel'\'' could not be installed
```

you can always use the `details` button to check the pipelines then go to the 
`pr-build-run` > `run-build`  

anyhow this is a issue from the base image, as ipykernel have deleted there version it seems 
i will try to fix it in the base image.

**Comment by harshad16** [comment]
oh this is not the issue of base image , it is correct 
the issue is here in the repo 
https://github.com/aicoe-aiops/ocp-ci-analysis/blob/1d602bdb28928e10c9584097c2c8e983b0b09fe5/Pipfile.lock#L269

from the ipykernel deleted 5.5.2 
https://pypi.org/project/ipykernel/#history
please updated the pipfile.lock

**Comment by harshad16** [comment]
asking the bot kebechet to help with this issue

File: notebooks/data-sources/TestGrid/metrics/test_pass_failures.ipynb [hunk 1]
```diff
@@ -2,91 +2,143 @@
  "cells": [
```
Comment (MichaelClifford) [inline notebooks/data-sources/TestGrid/metrics/test_pass_failures.ipynb:2]: <p>Do you still need #noqa here?</p><p> </p><pre class="ql-syntax" spellcheck="false">from ipynb.fs.defs.metric_template import (&nbsp;# noqa: E402
&nbsp;&nbsp;testgrid_labelwise_encoding,
&nbsp;&nbsp;CephCommunication,
&nbsp;&nbsp;save_to_disk,
&nbsp;&nbsp;read_from_disk,
)&nbsp;# noqa: E402
</pre><p> </p><p>I think this cell will pass pre-commit. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/200/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (MichaelClifford) [inline notebooks/data-sources/TestGrid/metrics/test_pass_failures.ipynb:2]: <p>Do you want to limit the data uploaded? Looks like there are 20 Million records, but then only pushing 1 million to ceph. Is that your intention here?</p><p> </p><pre class="ql-syntax" spellcheck="false">cc.upload_to_ceph(combined_df.head(1000000)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/200/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (hemajv) [reply notebooks/data-sources/TestGrid/metrics/test_pass_failures.ipynb:2]: Yes, the data is quite large to be uploaded to ceph so trying to store a smaller size

Comment (hemajv) [reply notebooks/data-sources/TestGrid/metrics/test_pass_failures.ipynb:2]: updated :+1:

**Review by Shreyanand — CHANGES_REQUESTED** [review]
Looks good! Slight change, if we add automation flag code like Michael added in [here](https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/208/) ,we could save some effort later :)

**Comment by hemajv** [comment]
> Looks good! Slight change, if we add automation flag code like Michael added in [here](https://app.reviewnb.com/aicoe-aiops/ocp-ci-analysis/pull/208/) ,we could save some effort later :)

@Shreyanand good point, updated :+1:

**Pull request merged**